lib/Sema/TypeCheckDeclObjC.cpp

```diff
@@ -985,40 +975,43 @@ bool swift::isRepresentableInObjC(const VarDecl *VD, ObjCReason Reason) {
     T = RST->getReferentType();
   }
   ASTContext &ctx = VD->getASTContext();
+  DiagnosticStateRAII diagState(ctx.Diags);
+
   bool Result = T->isRepresentableIn(ForeignLanguage::ObjectiveC,
                                      VD->getDeclContext());
-  bool Diagnose = shouldDiagnoseObjCReason(Reason, ctx);
+  auto behavior = behaviorLimitForObjCReason(Reason, ctx);
 
-  if (Result && checkObjCInExtensionContext(VD, Diagnose))
+  if (Result && checkObjCInExtensionContext(VD, behavior))
     return false;
 
   if (checkObjCInForeignClassContext(VD, Reason))
     return false;
   if (checkObjCActorIsolation(VD, Reason))
     return false;
 
-  if (!Diagnose || Result)
-    return Result;
-
-  SourceRange TypeRange = VD->getTypeSourceRangeForDiagnostics();
-  // TypeRange can be invalid; e.g. '@objc let foo = SwiftType()'
-  if (TypeRange.isInvalid())
-    TypeRange = VD->getNameLoc();
-
-  VD->diagnose(diag::objc_invalid_on_var, getObjCDiagnosticAttrKind(Reason))
-      .highlight(TypeRange);
-  diagnoseTypeNotRepresentableInObjC(VD->getDeclContext(),
-                                     VD->getInterfaceType(),
-                                     TypeRange);
-  describeObjCReason(VD, Reason);
+  if (!Result) {
+    SourceRange TypeRange = VD->getTypeSourceRangeForDiagnostics();
+    // TypeRange can be invalid; e.g. '@objc let foo = SwiftType()'
+    if (TypeRange.isInvalid())
+      TypeRange = VD->getNameLoc();
+
+    VD->diagnose(diag::objc_invalid_on_var, getObjCDiagnosticAttrKind(Reason))
+        .highlight(TypeRange)
+        .limitBehavior(behavior);
+    diagnoseTypeNotRepresentableInObjC(VD->getDeclContext(),
+                                       VD->getInterfaceType(),
+                                       TypeRange, behavior);
+    describeObjCReason(VD, Reason);
+  }
```

Comment:
Nit: Is the change in polarity here deliberate? The old code had an early exit, which helped reduce the indentation a bit. I see that there is a similar change in polarity in `swift::isRepresentableInObjC(const SubscriptDecl *SD, ObjCReason Reason)`.

Reply:
It's semi-deliberate. Partially it's just convenient to treat `behavior` as an opaque value that we pass around without examining, but I also think it's better to do it this way because there are fewer opportunities for us to accidentally skip over logic and get differences in logic between different `ObjCReason` values.